src/main/java/io/cdap/plugin/dlp/SensitiveRecordRedaction.java

```diff
@@ -0,0 +1,391 @@
+            ErrorConfig errorConfig = config.getErrorConfig("");
+            Gson gson = new Gson();
+
+            //Checking that custom template is defined if it is selected in one of the transforms
+            List<String> filters = Arrays.asList(config.getFilters());
+            if (!customTemplateEnabled && filters.contains("NONE")) {
+              collector.addFailure(String.format("This transform depends on custom template that was not defined.",
+                                                 config.getTransform(), String.join(", ", config.getFields())),
+                                   "Enable the custom template option and provide the name of it.")
+                .withConfigElement(FIELDS_TO_TRANSFORM, gson.toJson(errorConfig));
+            }
+            //Validate the config for the transform
+            config.validate(collector, inputSchema, FIELDS_TO_TRANSFORM);
+
+            //Check that custom template and built-in types are not mixed
+            anyTransformUsedCustomTemplate = anyTransformUsedCustomTemplate || filters.contains("NONE");
+            if (firstTransformUsedCustomTemplate == null) {
+              firstTransformUsedCustomTemplate = filters.contains("NONE");
+            } else {
+              if (filters.contains("NONE") != firstTransformUsedCustomTemplate) {
+                errorConfig.setTransformPropertyId("filters");
+                collector.addFailure("Cannot use custom templates and built-in filters in the same plugin instance.",
+                                     "All transforms must use custom templates or built-in filters, not a "
+                                       + "combination of both.")
+                  .withConfigElement(FIELDS_TO_TRANSFORM, gson.toJson(errorConfig));
+
+              }
+            }
+
+            // Make sure the combination of field, transform and filter are unique
+            for (String field : config.getFields()) {
+              for (String filter : config.getFilterDisplayNames()) {
+                String transformKey = String.format("%s:%s", field, filter);
+                if (transforms.containsKey(transformKey)) {
+
+                  String errorMessage;
+                  if (transforms.get(transformKey).equals(config.getTransform())) {
+                    errorMessage = String.format(
+                      "Combination of transform, filter and field must be unique. Found multiple definitions for '%s' "
+                        + "transform on '%s' with filter '%s'", config.getTransform(), field, filter);
+
+                  } else {
+                    errorMessage = String.format(
+                      "Only one transform can be defined per field and filter combination. Found conflicting transforms"
+                        + " '%s' and '%s'",
+                      transforms.get(transformKey), config.getTransform());
+
+                  }
+                  errorConfig.setTransformPropertyId("");
```

Comment:
To clear the field incase it was set by the pervious checks. This same object is used for the other validation steps (since most of it's properties do not change)